File: library/src/main/java/renetik/android/view/list/CSRowView.java
package renetik.android.view.list;

import android.view.View;

import renetik.android.viewbase.CSLayoutId;
import renetik.android.viewbase.CSView;

/**
 * Created by renetik on 08/12/17.
 */

public class CSRowView<T> extends CSView<View> {
    private T _row;

    public CSRowView(CSListController parent, CSLayoutId layout) {
        super(parent, layout);
    }

    public T data() {
        return _row;
    }

    public void data(T row) {
        _row = row;
        onLoad(_row);
    }

    protected void onLoad(T row) {
    }
}

File: library/src/main/java/renetik/android/view/list/CSRowView.kt
package renetik.android.view.list

import android.view.View

import renetik.android.view.base.CSLayoutId
import renetik.android.view.base.CSView

open class CSRowView<RowType : Any>(parent: CSListController<RowType, *>, layout: CSLayoutId) : CSView<View>(parent, layout) {

    lateinit var data: RowType

    fun data(row: RowType) {
        data = row
        onLoad(data)
    }

    protected open fun onLoad(row: RowType) = Unit
}
